sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1529,6 +1529,103 @@ def _df_getitem_unicode_idx_impl(self, idx):
     ty_checker.raise_exc(idx, expected_types, 'idx')
 
 
+def df_getitem_tuple_iat_codegen(self, row, col):
+    """
+    Example of generated implementation:
+        def _df_getitem_tuple_iat_impl(self, idx):
+            row, _ = idx
+            data = get_dataframe_data(self._dataframe, 1)
+            res_data = pandas.Series(data)
+            return res_data.iat[row]
+    """
+    func_lines = ['def _df_getitem_tuple_iat_impl(self, idx):',
+                  '  row, _ = idx']
+    if -1 < col < len(self.columns):
+        func_lines += [
+            f"  data = get_dataframe_data(self._dataframe, {col})",
+            f"  res_data = pandas.Series(data)",
+            f"  return res_data.iat[row]",
+        ]
+    else:
+        func_lines += ["  raise ValueError('Index is out of bounds for axis')"]
+
+    func_text = '\n'.join(func_lines)
+    global_vars = {'pandas': pandas,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def gen_df_getitem_tuple_iat_impl(self, row, col):
+    func_text, global_vars = df_getitem_tuple_iat_codegen(self, row, col)
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _reduce_impl = loc_vars['_df_getitem_tuple_iat_impl']
+
+    return _reduce_impl
+
+
+@sdc_overload(operator.getitem)
+def sdc_pandas_dataframe_accessor_getitem(self, idx):
+    if not isinstance(self, DataFrameGetitemAccessorType):
+        return None
+
+    accessor = self.accessor.literal_value
+
+    if accessor == 'iat':
+        if isinstance(idx, types.Tuple):
+            row = idx[0]
+            col = idx[1].literal_value
+            return gen_df_getitem_tuple_iat_impl(self.dataframe, row, col)
```

Comment:
I believe you don't need codegen here.
```python
def df_getitem_iat_tuple_impl(self, idx):
    row, _ = idx
    data = get_dataframe_data(self._dataframe, col)
    res_data = pandas.Series(data)

    return res_data.iat[row]    
```
should work just fine. And if it doesn't, you could use `literally`:
```python
def df_getitem_iat_tuple_impl(self, idx):
    row, _ = idx
    col_ = literally(col)
    data = get_dataframe_data(self._dataframe, col_)
    res_data = pandas.Series(data)

    return res_data.iat[row]    
```